src/main/java/io/cryostat/recordings/RecordingTargetHelper.java

```diff
@@ -137,7 +137,8 @@ public List<IRecordingDescriptor> getRecordings(ConnectionDescriptor connectionD
     }
 
     public IRecordingDescriptor startRecording(
-            boolean restart,
+            boolean restart, // Deprecated: Use replace parameter instead
+            String replace,
```

Comment:
I'd like this refactored so that the now-deprecated `restart` parameter is removed from the method signature(s), either by completely removing it or by adding a second overloaded method definition with the updated signature. I think ideally this method would have something like `ReplacementPolicy replace` instead of `boolean restart, String replace`, where `ReplacementPolicy` would be an enum. This should work pretty easily since effectively `restart = false` is the same as `ReplacementPolicy.NEVER` and `restart = true` is the same as `ReplacementPolicy.ALWAYS`, and `ReplacementPolicy.STOPPED` is the new behaviour.

Looking good otherwise.

Reply:
On this change, I just remember that I need to update the `types.graphqls` file. My question is 
``` java
input RecordingSettings {
    replace: String
    name: String!
    template: String!
    templateType: String!
    duration: Long
    continuous: Boolean
    toDisk: Boolean
    maxSize: Long
    maxAge: Long
    archiveOnStop: Boolean
    metadata: Object
}
```
do I leave it as string or change it to `replace: replacementPolicy`
@maxcao13 @andrewazores

Reply:
You probably want to create another Type in that `types.graphql` file called `ReplacementPolicy` with its relevant fields and use that.